### ISSUE TITLE
PPC64le image generation

### ISSUE THREAD

**Issue by boomanaiden154** [boomanaiden154/matplotlib-test-images#2]
Something a little fishy is going on with the PPC64LE image generation and more investigation is needed. After taking the images from the Github actions pipeline and trying to use them to test matplotlib, there are a whole host of image comparison failures that should not be happening.
```
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_agg_filter.py::test_agg_filter_alpha[png]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_agg.py::test_agg_filter[png]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_axes.py::test_imshow[png]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_axes.py::test_imshow_clip[png]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_axes.py::test_specgram_magnitude[png]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_axes.py::test_specgram_angle[png]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_image.py::test_nonuniform_and_pcolor[png]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_axes.py::test_specgram[png]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_image.py::test_imshow_masked_interpolation[png]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_image.py::test_rgba_antialias[png]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_colors.py::test_light_source_topo_surface[png]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_backend_svg.py::test_noscale[png]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_image.py::test_imshow_flatfield[png]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_image.py::test_image_clip[png]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_image.py::test_imshow_bignumbers[png]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_image.py::test_image_cliprect[png]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_image.py::test_imshow_bignumbers_real[png]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_image.py::test_imshow[png]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_image.py::test_no_interpolation_origin[png]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_image.py::test_bbox_image_inverted[png]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_image.py::test_image_composite_background[png]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_image.py::test_zoom_and_clip_upper_origin[png]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_constrainedlayout.py::test_colorbars_no_overlapH[png]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_image.py::test_image_composite_alpha[png]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_image.py::test_log_scale_image[png]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_constrainedlayout.py::test_colorbars_no_overlapV[png]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_image.py::test_rotate_image[png]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_image.py::test_mask_image_over_under[png]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_image.py::test_alpha_interp[png]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_image.py::test_figimage[png-False]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_image.py::test_figimage[png-True]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_image.py::test_mask_image[png]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_image.py::test_image_alpha[png]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_image.py::test_imshow_endianess[png]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_image.py::test_image_interps[png]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_patheffects.py::test_patheffect1[png]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_patheffects.py::test_patheffect2[png]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_png.py::test_pngsuite[png]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_pickle.py::test_complete[png]
FAILED test-env/lib/python3.10/site-packages/matplotlib-3.5.1-py3.10-linux-ppc64le.egg/matplotlib/tests/test_tightlayout.py::test_tight_layout5[png]
```

**Comment by boomanaiden154** [comment]
Alpine is currently building against powerpc v8 (-march=power8 in GCC). Need to check and make sure that this is getting applied when `python3 setup.py` builds the package. This might be causing some incompatibilities as without the flag gcc will try and build code that will work on all previous versions of ppc that are still supported.

**Comment by boomanaiden154** [comment]
This is an example of a current gcc command being used during the compile step on the Github Actions CI:
```
gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fomit-frame-pointer -g -fomit-frame-pointer -g -fomit-frame-pointer -g -DTHREAD_STACK_SIZE=0x200000 -fPIC -DMPL_DEVNULL=/dev/null -DPY_ARRAY_UNIQUE_SYMBOL=MPL_matplotlib__qhull_ARRAY_API -DNPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION -D__STDC_FORMAT_MACROS=1 -I/usr/lib/python3.10/site-packages/numpy/core/include -I/usr/include/python3.10 -c src/qhull_wrap.cpp -o build/temp.linux-ppc64le-3.10/matplotlib._qhull/src/qhull_wrap.o -fvisibility=hidden -flto
```
The corresponding line from the Aports CI build log:
```
gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fomit-frame-pointer -g -fomit-frame-pointer -g -fomit-frame-pointer -g -DTHREAD_STACK_SIZE=0x200000 -Os -fomit-frame-pointer -Os -fomit-frame-pointer -fPIC -DMPL_DEVNULL=/dev/null -DPY_ARRAY_UNIQUE_SYMBOL=MPL_matplotlib__qhull_ARRAY_API -DNPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION -D__STDC_FORMAT_MACROS=1 -I/usr/lib/python3.10/site-packages/numpy/core/include -I/usr/include/python3.10 -c src/qhull_wrap.cpp -o build/temp.linux-ppc64le-3.10/matplotlib._qhull/src/qhull_wrap.o -fvisibility=hidden -flto
```
Looks like the difference between the build commands is actually a couple extra repetitions of `-fomit-frame-pointer` in the Alpine CI compile as well as `-Os`. Will need to do some testing to see if disabling space saving optimizations actually fixes this problem. If so, disabling space saving optimizations on the Alpine side will be necessary and the ppc64le test patches can be removed.

**Comment by boomanaiden154** [comment]
Randomly fixed itself with the upgrade to 3.6.0. Not sure what ended up fixing the issue, but this can be closed now.
